chess_gui.py

```diff
@@ -174,23 +243,44 @@ def main():
                 elif e.key == py.K_u:
                     game_state.undo_move()
                     print(len(game_state.move_log))
-
+        if valid:
+            board = "\n---board---\n"
+           for row in range(8):
+            for col in range(8):
+                if game_state.is_valid_piece(row, col):
+                    board += game_state.get_piece(row, col).get_name()
+                        board += " "
+                    else:
+                        board += Player.EMPTY
+                        board += " "
+                board += '\n'
```

Comment:
You use this for loop twice, consider separating it to a different function

Reply:
While writing the code I already tried because the double code bothered me and for some reason it didn't work - I'll try again

Reply:
I made the changes in the repository itself. If there is a problem with the exercise, let me know and upload the code I have on my computer.